airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlCdcTargetPosition.java

```diff
@@ -77,17 +81,30 @@ public int hashCode() {
 
   public static MssqlCdcTargetPosition getTargetPosition(final JdbcDatabase database, final String dbName) {
     try {
-      final List<JsonNode> jsonNodes = database
-          .bufferedResultSetQuery(connection -> connection.createStatement().executeQuery(
-              "USE [" + dbName + "]; SELECT sys.fn_cdc_get_max_lsn() AS max_lsn;"), JdbcUtils.getDefaultSourceOperations()::rowToJson);
+      // We might have to wait a bit before querying the max_lsn to give the CDC capture job
+      // a chance to catch up. This is important in tests, where reads might occur in quick succession
+      // which might leave the CT tables (which Debezium consumes) in a stale state.
+      final JsonNode sourceConfig = database.getSourceConfig();
+      final Duration delay = (sourceConfig != null && sourceConfig.has("is_test") && sourceConfig.get("is_test").asBoolean())
+          ? MAX_LSN_QUERY_DELAY_TEST
+          : MAX_LSN_QUERY_DELAY;
+      final String maxLsnQuery = """
+                                 USE [%s];
+                                 WAITFOR DELAY '%02d:%02d:%02d';
```

Comment:
oh this is ugly! who had the genius idea of forcing a HH:MM:SS format here ???

Reply:
Bill Gates, most likely. Or someone who doesn't care about ISO 8601.